server/lib/src/be/mod.rs

```diff
@@ -579,6 +564,69 @@ pub trait BackendTransaction {
         })
     }
 
+    fn filter2idl_sub(
+        &mut self,
+        attr: &AttrString,
+        sub_idx_key: String,
+    ) -> Result<(IdList, FilterPlan), OperationError> {
+        // Now given that idx_key, we will iterate over the possible graphemes.
+        let mut grapheme_iter = trigraph_iter(&sub_idx_key);
+
+        // Substrings are always partial because we have to split the keys up
+        // and we don't pay attention to starts/ends with conditions. We need
+        // the caller to check those conditions manually at run time. This lets
+        // the index focus on trigraph indexes only rather than needing to
+        // worry about those other bits. In a way substring indexes are "fuzzy".
+
+        let mut idl = match grapheme_iter.next() {
+            Some(idx_key) => {
+                match self
+                    .get_idlayer()
+                    .get_idl(attr, IndexType::SubString, &idx_key)?
```

Comment:
Clippy says to do this
```suggestion
                    .get_idl(attr, IndexType::SubString, idx_key)?
```